courses/models.py

```diff
@@ -273,12 +272,7 @@ def has_exam(self):
         """
         Check if the user is authorized for an exam for a course run
         """
-        return (
-            self.edx_course_key and
-            self.course and
-            not is_blank(self.course.exam_module) and
-            not is_blank(self.course.program.exam_series_code)
-        )
+        return self.course.exam_runs.filter(date_last_eligible__gt=datetime.now(pytz.utc)).exists()
```

Comment:
`exam_series_code` isn't nullible for the `ExamRun`, so that shouldn't be an issue. Not sure what you been by the coruse key, because we don't handle that with exams.